specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json

```diff
@@ -75,6 +129,51 @@
       },
       "description": "Tenant access information update parameters of the API Management service"
     },
+    "TenantSettingsCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TenantSettingsContract"
+          },
+          "description": "Page values.",
+          "readOnly": true
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any.",
+          "readOnly": true
+        }
+      },
+      "description": "Paged AccessInformation list representation."
+    },
+    "TenantSettingsContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TenantSettingsContractProperties",
+          "description": "TenantSettings entity contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
+      ],
+      "description": "Tenant Settings."
+    },
+    "TenantSettingsContractProperties": {
+      "properties": {
+        "settings": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
having a generic property bag instead of a defined schema means that policy aliases cannot be generated, and hence you cannot create/enforce policies on these settings.

Reply:
instead consider using the top-level kind property as a discriminator to determine the schema of the settings, and also consider making the schema hierarchical instead of flattening the way you have in the example.

Reply:
We don't have predefined settings, customers may introduce their own settings, that why we use Dictionary<string, string>

Reply:
that is fine as long as you understand the implication to policy https://armwiki.azurewebsites.net/api_contracts/guidelines/policy.html#plcy006-follow-the-right-semantics-for-put-and-patch-operations - if you ever need to enforce policies on these settings, you would have to define a concrete schema i.e. something like an array object where each item has a "name" and "value"

Reply:
Will be implemented in separate task: https://msazure.visualstudio.com/One/_workitems/edit/9030574  
The current PR is for renaming endpoint only, and should be merged asap since the implementation is already in production.